pkg/file/delete.go

```diff
@@ -44,6 +44,15 @@ func (r renamerRemoverImpl) Stat(path string) (fs.FileInfo, error) {
 	return r.StatFn(path)
 }
 
+func newRenamerRemoverImpl() renamerRemoverImpl {
+	return renamerRemoverImpl{
+		RenameFn:    os.Rename,
```

Comment:
Depending on the scope of this PR/implementation `os.Rename` might not be enough.
There 2 issues with using it directly
- It doesnt rename files when the source and the destination path are in diferrent filesystems/disks.
Relevant links [original issue](https://github.com/golang/go/issues/13766), [error detection sample](https://github.com/Masterminds/glide/pull/587/files). From a quick look it seems that this behaviour was by design and the most common workaround is to detect the error and then copy and delete the file.
- Rename overwrites existing files

A quick and dirty [code sample](https://go.dev/play/p/Fyh2OcM0Nln) demonstrating the issues
The second issue is pretty easy to handle from a stash plugin so it may or may not be a problem provided we mention that somewhere?
The first is a little more complex and imho it would be better handled by stash.